### ISSUE TITLE
Getting an error when run the WSO2 Identity Server 5.12 Alpha9 pack in Windows 10 Pro edition.

### ISSUE THREAD

**Issue by Nilki** [wso2/product-is#12767]
When run the WSO2 Identity Server 5.12 pack in Windows 10 Pro edition, 20H2 version I am getting an error in Deployed webapps, webapps are not deploying in it.

```
[2021-11-10 18:09:19,174] [] ERROR {org.wso2.carbon.webapp.mgt.loader.SharedClassLoaderFactory} - Error while creating class loader for CXF3 java.io.IOException: The filename, directory name, or volume label syntax is incorrect
        at java.io.WinNTFileSystem.canonicalize0(Native Method)
        at java.io.WinNTFileSystem.canonicalize(WinNTFileSystem.java:428)
        at java.io.File.getCanonicalPath(File.java:624)
        at java.io.File.getCanonicalFile(File.java:649)
        at org.apache.catalina.startup.ClassLoaderFactory.createClassLoader(ClassLoaderFactory.java:174)
        at org.wso2.carbon.webapp.mgt.loader.SharedClassLoaderFactory.createClassloader(SharedClassLoaderFactory.java:97)
        at org.wso2.carbon.webapp.mgt.loader.SharedClassLoaderFactory.lambda$init$0(SharedClassLoaderFactory.java:67)
        at java.util.TreeMap.forEach(TreeMap.java:1005)
        at org.wso2.carbon.webapp.mgt.loader.SharedClassLoaderFactory.init(SharedClassLoaderFactory.java:66)
        at org.wso2.carbon.webapp.mgt.loader.ClassloadingContextBuilder.buildSystemConfig(ClassloadingContextBuilder.java:73)
        at org.wso2.carbon.webapp.mgt.loader.WebappClassloadingContext.<clinit>(WebappClassloadingContext.java:56)
        at org.wso2.carbon.webapp.mgt.loader.ClassloadingContextBuilder.buildClassloadingContext(ClassloadingContextBuilder.java:78)
        at org.wso2.carbon.webapp.mgt.loader.CarbonWebappLoader.startInternal(CarbonWebappLoader.java:45)
        at org.apache.catalina.util.LifecycleBase.start(LifecycleBase.java:183)
        at org.apache.catalina.core.StandardContext.startInternal(StandardContext.java:5086)
        at org.apache.catalina.util.LifecycleBase.start(LifecycleBase.java:183)
        at org.apache.catalina.core.ContainerBase.addChildInternal(ContainerBase.java:726)
        at org.apache.catalina.core.ContainerBase.addChild(ContainerBase.java:698)
        at org.apache.catalina.core.StandardHost.addChild(StandardHost.java:696)
        at org.wso2.carbon.tomcat.internal.CarbonTomcat.addWebApp(CarbonTomcat.java:303)
        at org.wso2.carbon.tomcat.internal.CarbonTomcat.addWebApp(CarbonTomcat.java:209)
        at org.wso2.carbon.webapp.mgt.TomcatGenericWebappsDeployer.handleWebappDeployment(TomcatGenericWebappsDeployer.java:250)
        at org.wso2.carbon.webapp.mgt.TomcatGenericWebappsDeployer.handleExplodedWebappDeployment(TomcatGenericWebappsDeployer.java:238)
        at org.wso2.carbon.webapp.mgt.TomcatGenericWebappsDeployer.handleHotDeployment(TomcatGenericWebappsDeployer.java:168)
        at org.wso2.carbon.webapp.mgt.TomcatGenericWebappsDeployer.deploy(TomcatGenericWebappsDeployer.java:135)
        at org.wso2.carbon.webapp.mgt.AbstractWebappDeployer.deployThisWebApp(AbstractWebappDeployer.java:106)
        at org.wso2.carbon.webapp.mgt.AbstractWebappDeployer.deploy(AbstractWebappDeployer.java:96)
        at org.wso2.carbon.webapp.deployer.WebappDeployer.deploy(WebappDeployer.java:42)
        at org.apache.axis2.deployment.repository.util.DeploymentFileData.deploy(DeploymentFileData.java:136)
        at org.apache.axis2.deployment.DeploymentEngine.doDeploy(DeploymentEngine.java:807)
        at org.apache.axis2.deployment.repository.util.WSInfoList.update(WSInfoList.java:144)
        at org.apache.axis2.deployment.RepositoryListener.update(RepositoryListener.java:377)
        at org.apache.axis2.deployment.RepositoryListener.checkServices(RepositoryListener.java:254)
        at org.apache.axis2.deployment.DeploymentEngine.loadServices(DeploymentEngine.java:135)
        at org.wso2.carbon.core.CarbonAxisConfigurator.deployServices(CarbonAxisConfigurator.java:568)
        at org.wso2.carbon.core.internal.DeploymentServerStartupObserver.completingServerStartup(DeploymentServerStartupObserver.java:51)
        at org.wso2.carbon.core.internal.CarbonCoreServiceComponent.notifyBefore(CarbonCoreServiceComponent.java:258)
        at org.wso2.carbon.core.internal.StartupFinalizerServiceComponent.completeInitialization(StartupFinalizerServiceComponent.java:166)
        at org.wso2.carbon.core.internal.StartupFinalizerServiceComponent.serviceChanged(StartupFinalizerServiceComponent.java:323)
        at org.eclipse.osgi.internal.serviceregistry.FilteredServiceListener.serviceChanged(FilteredServiceListener.java:113)
        at org.eclipse.osgi.internal.framework.BundleContextImpl.dispatchEvent(BundleContextImpl.java:985)
        at org.eclipse.osgi.framework.eventmgr.EventManager.dispatchEvent(EventManager.java:234)
        at org.eclipse.osgi.framework.eventmgr.ListenerQueue.dispatchEventSynchronous(ListenerQueue.java:151)
        at org.eclipse.osgi.internal.serviceregistry.ServiceRegistry.publishServiceEventPrivileged(ServiceRegistry.java:866)
        at org.eclipse.osgi.internal.serviceregistry.ServiceRegistry.publishServiceEvent(ServiceRegistry.java:804)
        at org.eclipse.osgi.internal.serviceregistry.ServiceRegistrationImpl.register(ServiceRegistrationImpl.java:130)
        at org.eclipse.osgi.internal.serviceregistry.ServiceRegistry.registerService(ServiceRegistry.java:228)
        at org.eclipse.osgi.internal.framework.BundleContextImpl.registerService(BundleContextImpl.java:525)
        at org.eclipse.osgi.internal.framework.BundleContextImpl.registerService(BundleContextImpl.java:544)
        at org.wso2.carbon.server.admin.internal.ServerAdminServiceComponent.activate(ServerAdminServiceComponent.java:99)
        at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
        at sun.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
        at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
        at java.lang.reflect.Method.invoke(Method.java:498)
        at org.eclipse.equinox.internal.ds.model.ServiceComponent.activate(ServiceComponent.java:260)
        at org.eclipse.equinox.internal.ds.model.ServiceComponentProp.activate(ServiceComponentProp.java:146)
        at org.eclipse.equinox.internal.ds.model.ServiceComponentProp.build(ServiceComponentProp.java:345)
        at org.eclipse.equinox.internal.ds.InstanceProcess.buildComponent(InstanceProcess.java:620)
        at org.eclipse.equinox.internal.ds.InstanceProcess.buildComponents(InstanceProcess.java:197)
        at org.eclipse.equinox.internal.ds.Resolver.getEligible(Resolver.java:343)
        at org.eclipse.equinox.internal.ds.SCRManager.serviceChanged(SCRManager.java:222)
        at org.eclipse.osgi.internal.serviceregistry.FilteredServiceListener.serviceChanged(FilteredServiceListener.java:113)
        at org.eclipse.osgi.internal.framework.BundleContextImpl.dispatchEvent(BundleContextImpl.java:985)
        at org.eclipse.osgi.framework.eventmgr.EventManager.dispatchEvent(EventManager.java:234)
        at org.eclipse.osgi.framework.eventmgr.ListenerQueue.dispatchEventSynchronous(ListenerQueue.java:151)
        at org.eclipse.osgi.internal.serviceregistry.ServiceRegistry.publishServiceEventPrivileged(ServiceRegistry.java:866)
        at org.eclipse.osgi.internal.serviceregistry.ServiceRegistry.publishServiceEvent(ServiceRegistry.java:804)
        at org.eclipse.osgi.internal.serviceregistry.ServiceRegistrationImpl.register(ServiceRegistrationImpl.java:130)
        at org.eclipse.osgi.internal.serviceregistry.ServiceRegistry.registerService(ServiceRegistry.java:228)
        at org.eclipse.osgi.internal.framework.BundleContextImpl.registerService(BundleContextImpl.java:525)
        at org.eclipse.osgi.internal.framework.BundleContextImpl.registerService(BundleContextImpl.java:544)
        at org.wso2.carbon.core.init.CarbonServerManager.initializeCarbon(CarbonServerManager.java:529)
        at org.wso2.carbon.core.init.CarbonServerManager.removePendingItem(CarbonServerManager.java:305)
        at org.wso2.carbon.core.init.PreAxis2ConfigItemListener.bundleChanged(PreAxis2ConfigItemListener.java:118)
        at org.eclipse.osgi.internal.framework.BundleContextImpl.dispatchEvent(BundleContextImpl.java:973)
        at org.eclipse.osgi.framework.eventmgr.EventManager.dispatchEvent(EventManager.java:234)
        at org.eclipse.osgi.framework.eventmgr.EventManager$EventThread.run(EventManager.java:345)
[2021-11-10 18:09:19,420] []  INFO {org.wso2.carbon.webapp.mgt.TomcatGenericWebappsDeployer} - Deployed webapp: StandardEngine[Catalina].StandardHost[localhost].StandardContext[/STRATOS_ROOT].File[D:\Nilki\WSO2\IdentityServer\wso2is-5.12.0-alpha9\bin\..\repository\deployment\server\webapps\STRATOS_ROOT]
[2021-11-10 18:09:19,718] [] ERROR {org.apache.catalina.core.ContainerBase.[Catalina].[localhost].[/accountrecoveryendpoint]} - Exception sending context initialized event to listener instance of class [org.wso2.carbon.identity.mgt.endpoint.util.listener.IdentityManagementEndpointContextListener] java.lang.NoClassDefFoundError: org/apache/cxf/jaxrs/provider/json/JSONProvider
```

 - Product Version: WSO2 Identity Server 5.12.0 Alpha9
 - OS: Windows 10 Pro edition, 20H2 version
 - Java Version: oracle 1.8.0_291

**Comment by Kogula04** [comment]
I also getting the same error when run the WSO2 Identity Server 5.12.0 alpha6 and WSO2 Identity Server 5.12.0 alpha7 in Windows 10 Pro edition, 20H2 version.

**Comment by Nirmal-De-Silva-99** [comment]
I'm also getting the same error when running the WSO2 Identity Server 5.12.0 alpha9 in Windows 10 Pro edition, 20H2 version.

**Comment by MawliDeSilva** [comment]
I also face the same error when running WSO2 Identity Server 5.12.0 alpha9 in Windows 10 Pro edition, 20H2 version.

**Comment by induwaram** [comment]
I'm getting the same error as well with WSO2 Identity Server 5.12.0 alpha9 in Windows 10 Pro edition, 20H2 version.

**Comment by CharinduLiyanage** [comment]
I'm getting the same error as well with WSO2 Identity Server 5.12.0 alpha9 in Windows 10 Pro edition, 20H2 version.

**Comment by randiepathirage** [comment]
I also get the same error when running WSO2 Identity Server 5.12.0 alpha9 in Windows 10 Pro edition, 20H2 version.

**Comment by randiepathirage** [comment]
This error occurs due to the fail when deploying webapps. Changing CXF3 Classpath as follows     in
``` <IS_HOME>\repository\conf\tomcat\webapp-classloading-enviroments.xml ``` resolved this issue
```     <ExclusiveEnvironments>
              <ExclusiveEnvironment>
                    <Name>CXF3</Name>
                    <Classpath>${carbon.home}/lib/runtimes/cxf3/*.jar</Classpath>
             </ExclusiveEnvironment>
        </ExclusiveEnvironments> 
``` 
Apache Maven version: 3.8.3
Java version: 1.8.0_271
OS: Windows 10 Pro edition

**Comment by piyarathnalakmali** [comment]
The reported issue is introduced in [IS 5.12.0-m13](https://github.com/wso2/product-is/releases/tag/v5.12.0-m13) and reproducible in all the newer versions up to now. Web apps are deployed without any issues in [IS 5.12.0-m12](https://github.com/wso2/product-is/releases/tag/v5.12.0-m12). 
org.wso2.carbon.webapp.mgt is the component which is giving the error and there has been a version upgrade of that component between the working and broken IS releases. 

- IS 5.12.0-m12 - 4.10.4
- IS 5.12.0-m13 - 4.12.1

The web app classloading hierarchy is changed in between the above mentioned versions by this PR [carbon-deployment #347](https://github.com/wso2/carbon-deployment/pull/347). 
CXF3 class paths in `<IS_HOME>\repository\conf\tomcat\webapp-classloading-enviroments.xml` which is mentioned in the above workaround are not changed in the two versions and only how they are loaded is being modified.

**Comment by nilasini** [comment]
The fix for the issue will be available from IS-5.12.0-alpha16
